pkg/controller/statusmanager/pod_status.go

```diff
@@ -120,17 +128,47 @@ func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bo
 		Reason:             reason,
 		Message:            message,
 	}
-	SetNodeCondition(&node.Status.Conditions, condition)
-	err = status.client.Status().Update(context.TODO(), node)
-	if err != nil {
-		log.Error(err, "Failed to update node condition")
+	changed := SetNodeCondition(&node.Status.Conditions, condition)
+	if changed {
+		if ready {
+			last := time.Now()
+			_, present := sharedInfo.LastNetworkAvailabe[nodeName]
+			if !present {
+				sharedInfo.LastNetworkAvailabe[nodeName] = last
+				log.Info(fmt.Sprintf("Last network unavailable time of node %s is set to %s", nodeName, last))
+
+				go func() {
+					start := time.Now()
+					time.Sleep(time.Duration(operatortypes.TimeBeforeRecoverNetwork) * time.Second)
+					if (sharedInfo.LastNetworkAvailabe[nodeName]) != last {
+						log.Info(fmt.Sprintf("Last network unavailable time of node %s changed, expecting %v, got %v, goroutine %v exiting", nodeName, last, sharedInfo.LastNetworkAvailabe[nodeName], start))
+						return
+					}
+					err = status.client.Status().Update(context.TODO(), node)
+					if err != nil {
+						log.Error(err, "Failed to update node condition to true")
+					} else {
+						log.Info("Updated node condition to true")
+					}
+					return
+				}()
+			}
+
+		} else {
+			err = status.client.Status().Update(context.TODO(), node)
+			if err != nil {
+				log.Error(err, "Failed to update node condition to false")
+			} else {
+				log.Info("Updated node condition to false")
+			}
+		}
 	} else {
-		log.Info("Updated node condition")
```

Comment:
I think we can put this `else` in front.
```
if not changed {
	log.Info("Node condition is not changed")
}
```

Reply:
ok, done